variants/BBCmicrobitV2/variant.cpp

```diff
@@ -0,0 +1,58 @@
+/* Pins for micro:bit V2 */
+
+#include "variant.h"
+
+const uint32_t g_ADigitalPinMap[] = {
```

Comment:
Do you have a link to the pin out diagram of the board? (Just want to double check this).

Reply:
https://tech.microbit.org/hardware/schematic/

I don't believe there are any constraints on the ordering of the internal pins (i.e. user pin numbers > 20) so I've ordered them in a way that seems to make sense and be convenient for users. For example, the LED matrix pins for rows 1-5 are 21-25, so the second digit matches the row number.

I'd like the logo pin, speaker pin, and microphone power pin to follow the ROW pins immediately so MicroBlocks can use a simple comparison to check the range of user manipulable pins. (We don't want people to accidentally access the internal i2c or RX/TX pins since doing that would interfere with other functions. Also, they shouldn't use the microphone as a digital output since that could damage the microphone chip. They should only read the microphone as an analog input (A6). So all of those special pins should be located above the last manipulable pin (pin 28, RUN_MIC in the current pin map).

I also have a preliminary schematic but that was shared with me under NDA. If you want it, I'm sure Mark Williams or Jonny would send it to you.